src/leveldbwrapper.cpp

```diff
@@ -87,3 +106,18 @@ bool CLevelDBWrapper::WriteBatch(CLevelDBBatch& batch, bool fSync) throw(leveldb
     HandleError(status);
     return true;
 }
+
+// Prefixed with null character to avoid collisions with other keys
+const std::string CLevelDBWrapper::OBFUSCATE_KEY_KEY = "\000obfuscate_key";
+
+const unsigned int CLevelDBWrapper::OBFUSCATE_KEY_NUM_BYTES = 8;
+
+/**
+ * Returns a string (consisting of 8 random bytes) suitable for use as an 
+ * obfuscating XOR key. 
+ */
+std::string CLevelDBWrapper::CreateObfuscateKey() const {
+    unsigned char buff[OBFUSCATE_KEY_NUM_BYTES];
+    GetRandBytes(buff, OBFUSCATE_KEY_NUM_BYTES);
+    return std::string(&buff[0], &buff[OBFUSCATE_KEY_NUM_BYTES]);
```

Comment:
It's 7, a range constructor. Note that the second argument points to the position after the last element in the range. Pointers can be used like iterators, and it should be fine.

https://ideone.com/tBWxwM